scanpy/readwrite.py

```diff
@@ -277,15 +277,44 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = {'obs_names': dsets['barcodes'].astype(str)}
+            var_dict = {'var_names': dsets['name'].astype(str)}
+
+            if 'gene_id' not in dsets:
+                # Read metadata specific to a feature-barcode matrix
+                var_dict['gene_ids'] = dsets['id'].astype(str)
+            else:
+                # Read metadata specific to a probe-barcode matrix
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['gene_id'].astype(str),
+                        'probe_ids': dsets['id'].astype(str),
+                    }
+                )
+            var_dict['feature_types'] = dsets['feature_type'].astype(str)
+            if 'filtered_barcodes' in f['matrix']:
+                obs_dict['filtered_barcodes'] = dsets['filtered_barcodes'].astype(bool)
+
+            if 'features' in f['matrix']:
+                var_dict.update(
+                    (x, dsets[x].astype(bool if y.dtype.kind == 'b' else str))
+                    for x, y in f['matrix']['features'].items()
```

Comment:
Could we please use more explicit naming instead of "x" and "y"?

Reply:
Fixed.